--- core/modules/locale/src/LocaleFetch.php
+++ core/modules/locale/src/LocaleFetch.php
@@ -317,9 +317,19 @@
 
     // Check the status of local translation files.
     if (isset($source->files[LOCALE_TRANSLATION_LOCAL])) {
-      if ($file = $this->localeSource->sourceCheckFile($source)) {
+      $file = $this->localeSource->sourceCheckFile($source);
+      if ($file !== FALSE) {
         $this->localeSource->saveSource($source->name, $source->langcode, LOCALE_TRANSLATION_LOCAL, $file);
       }
+      else {
+        // try sites/all/translations as well
+        $source->files['local']->directory = "sites/all/translations";
+        $source->files['local']->uri = $source->files['local']->directory . $source->files['local']->filename;
+        $file = $this->localeSource->sourceCheckFile($source);
+        if ($file !== FALSE) {
+          $this->localeSource->saveSource($source->name, $source->langcode, LOCALE_TRANSLATION_LOCAL, $file);
+        }
+      }
       $checked = TRUE;
     }
