--- core/modules/field_ui/src/Element/FieldUiTable.php	2025-06-30 20:34:41
+++ core/modules/field_ui/src/Element/FieldUiTable.php	2025-06-30 20:37:09
@@ -228,7 +228,7 @@
    */
   public static function reduceOrder($array, $a) {
     $array = $array ?: [];
-    if (!empty($a['name'])) {
+    if (array_key_exists('name', $a) && !empty($a['name'])) {
       $array[] = $a['name'];
     }
     if (!empty($a['children'])) {
