--- core/modules/locale/src/LocaleImportBatch.php
+++ core/modules/locale/src/LocaleImportBatch.php
@@ -298,6 +298,11 @@
   public function batchFinished(bool $success, array $results): void {
     $logger = ($this->logger)();
     if ($success) {
+      try {
+        refreshAPICBlocks();
+      } catch(\Exception $e) {
+        \Drupal::logger('locale_err')->log('error', 'Error in refreshAPICBlocks()');
+      }
       $additions = $updates = $deletes = $skips = 0;
