--- core/lib/Drupal/Core/Theme/ThemeManager.php	2026-05-06 15:19:19
+++ core/lib/Drupal/Core/Theme/ThemeManager.php	2026-05-15 12:45:49
@@ -313,7 +313,7 @@
       // Invoke preprocess callbacks that are not in the invoke map, such as
       // those from an alter hook.
       elseif (is_callable($preprocessor_function)) {
-        call_user_func_array($preprocessor_function, [&$variables, $hook, $info]);
+        call_user_func_array($preprocessor_function, [&$variables, $hook, &$info]);
       }
       return $variables;
     };
