--- core/modules/node/src/NodeAccessRebuild.php	2025-01-06 22:41:17
+++ core/modules/node/src/NodeAccessRebuild.php	2025-01-08 14:35:57
@@ -147,6 +147,8 @@
     if (!isset($batch_builder)) {
       $this->messenger->addStatus($this->t('Content permissions have been rebuilt.'));
       $this->setNeedsRebuild(FALSE);
+    } else {
+      $this->messenger->addStatus($this->t('Content permissions rebuild has been initiated.'));
     }
   }
 
