--- core/modules/locale/src/StreamWrapper/TranslationsStream.php	2025-07-12 22:28:03
+++ core/modules/locale/src/StreamWrapper/TranslationsStream.php	2025-07-12 22:37:25
@@ -4,7 +4,7 @@
 
 use Drupal\Core\File\FileSystemInterface;
 use Drupal\Core\Site\Settings;
-use Drupal\Core\StreamWrapper\LocalStream;
+use Drupal\Core\StreamWrapper\PrivateStream;
 use Drupal\Core\StreamWrapper\StreamWrapperInterface;
 use Drupal\Core\StringTranslation\StringTranslationTrait;
 
@@ -13,7 +13,7 @@
  *
  * Provides support for storing translation files.
  */
-class TranslationsStream extends LocalStream {
+class TranslationsStream extends PrivateStream {
 
   use StringTranslationTrait;
 
@@ -45,16 +45,6 @@
     return static::basePath();
   }
 
-  /**
-   * phpcs:ignore Drupal.Files.LineLength
-   * Implements Drupal\Core\StreamWrapper\StreamWrapperInterface::getExternalUrl().
-   *
-   * @throws \LogicException
-   *   PO files URL should not be public.
-   */
-  public function getExternalUrl() {
-    throw new \LogicException('PO files URL should not be public.');
-  }
 
   /**
    * Returns the base path for translations://.
