--- core/modules/update/update.module
+++ core/modules/update/update.module
@@ -10,6 +10,19 @@
 use Drupal\update\Hook\UpdateThemeHooks;
 use Drupal\update\UpdateFetcherInterface;
 use Drupal\update\UpdateManagerInterface;
+/**
+ * Resolves if the current user can access updater menu items.
+ *
+ * It both enforces the 'administer software updates' permission and the global
+ * kill switch for the authorize.php script.
+ *
+ * @return bool
+ *   TRUE if the current user can access the updater menu items; FALSE
+ *   otherwise.
+ */
+function _update_manager_access() {
+  return Settings::get('allow_authorize_operations', TRUE) && \Drupal::currentUser()->hasPermission('administer software updates');
+}
 
 /**
  * Returns a warning message when there is no data about available updates.
@@ -181,7 +194,7 @@
   switch ($msg_reason) {
     case UpdateManagerInterface::NOT_SECURE:
       if ($msg_type == 'core') {
-        $text = t('There is a security update available for your version of Drupal. To ensure the security of your server, you should update immediately!', [], ['langcode' => $langcode]);
+        $text = t('There are updates available for your version of Drupal. To ensure the proper functioning of your site, you should update as soon as possible.', [], ['langcode' => $langcode]);
       }
       else {
         $text = t('There are security updates available for one or more of your modules or themes. To ensure the security of your server, you should update immediately!', [], ['langcode' => $langcode]);
@@ -378,8 +391,6 @@
  * @see https://www.drupal.org/node/3522119
  */
 function _update_manager_unique_identifier() {
-  @trigger_error(__FUNCTION__ . '() is deprecated in drupal:11.2.0 and is removed from drupal:12.0.0. There is no replacement. Use composer to manage the code for your site. See https://www.drupal.org/node/3522119', E_USER_DEPRECATED);
-
   static $id;
   if (!isset($id)) {
     $id = substr(hash('sha256', Settings::getHashSalt()), 0, 8);
@@ -404,8 +415,6 @@
  * @see https://www.drupal.org/node/3522119
  */
 function _update_manager_extract_directory($create = TRUE) {
-  @trigger_error(__FUNCTION__ . '() is deprecated in drupal:11.2.0 and is removed from drupal:12.0.0. There is no replacement. Use composer to manage the code for your site. See https://www.drupal.org/node/3522119', E_USER_DEPRECATED);
-
   static $directory;
   if (!isset($directory)) {
     $directory = 'temporary://update-extraction-' . _update_manager_unique_identifier();
@@ -433,8 +442,6 @@
  * @see https://www.drupal.org/node/3522119
  */
 function _update_manager_cache_directory($create = TRUE) {
-  @trigger_error(__FUNCTION__ . '() is deprecated in drupal:11.2.0 and is removed from drupal:12.0.0. There is no replacement. Use composer to manage the code for your site. See https://www.drupal.org/node/3522119', E_USER_DEPRECATED);
-
   static $directory;
   if (!isset($directory)) {
     $directory = 'temporary://update-cache-' . _update_manager_unique_identifier();
