--- core/modules/locale/src/LocaleProjectRepository.php	2025-06-30 20:47:11
+++ core/modules/locale/src/LocaleProjectRepository.php	2025-06-30 20:51:32
@@ -200,6 +200,9 @@
     $project_info = new ProjectInfo();
     $project_info->processInfoList($projects, $module_data, 'module', TRUE, $additional_allow_list);
     $project_info->processInfoList($projects, $theme_data, 'theme', TRUE, $additional_allow_list);
+    // Also add disabled modules in the list of projects to install
+    $project_info->processInfoList($projects, $module_data, 'module', FALSE, $additional_allow_list);
+    $project_info->processInfoList($projects, $theme_data, 'theme', FALSE, $additional_allow_list);
 
     // Allow other modules to alter projects before fetching and comparing.
     $this->moduleHandler->alter('locale_translation_projects', $projects);
--- core/modules/locale/src/LocaleProjectChecker.php	2025-06-30 20:47:11
+++ core/modules/locale/src/LocaleProjectChecker.php	2025-06-30 20:51:32
@@ -86,6 +86,18 @@
         if ($file) {
           $this->localeSource->saveSource($project, $langcode, LOCALE_TRANSLATION_LOCAL, $file);
         }
+        else {
+          // Try sites/all/translations as a fallback directory.
+          if (isset($source->files[LOCALE_TRANSLATION_LOCAL])) {
+            $local_file = $source->files[LOCALE_TRANSLATION_LOCAL];
+            $local_file->directory = 'sites/all/translations';
+            $local_file->uri = $local_file->directory . '/' . $local_file->filename;
+            $fallback_file = $this->localeSource->sourceCheckFile($source);
+            if ($fallback_file) {
+              $this->localeSource->saveSource($project, $langcode, LOCALE_TRANSLATION_LOCAL, $fallback_file);
+            }
+          }
+        }
       }
     }
   }
