--- core/modules/user/user.module	2025-07-01 19:14:39
+++ core/modules/user/user.module	2025-07-01 19:19:18
@@ -69,7 +69,15 @@
   @trigger_error(__FUNCTION__ . '() is deprecated in drupal:11.4.0 and is removed from drupal:13.0.0. Use entityTypeManager()->getStorage("user")->loadByProperties() instead. See https://www.drupal.org/node/3555936', E_USER_DEPRECATED);
   $users = \Drupal::entityTypeManager()->getStorage('user')
     ->loadByProperties(['name' => $name]);
-  return $users ? reset($users) : FALSE;
+  if (sizeof($users) > 1) {
+    throw new \Exception(sprintf('Attempt to load user by name (%s) returned multiple responses (%d). Unable to continue please contact your system administrator', $name, sizeof($users)));
+  }
+  else if (sizeof($users) === 1) {
+    return reset($users);
+  }
+  else {
+    return FALSE;
+  }
 }
 
 /**
@@ -551,11 +559,12 @@
 function user_role_revoke_permissions($rid, array $permissions = []): void {
   // Revoke permissions for the role.
   $role_storage = \Drupal::entityTypeManager()->getStorage('user_role');
-  $role = $role_storage->loadOverrideFree($rid);
-  foreach ($permissions as $permission) {
-    $role->revokePermission($permission);
+  if ($role = $role_storage->loadOverrideFree($rid)) {
+    foreach ($permissions as $permission) {
+      $role->revokePermission($permission);
+    }
+    $role->save();
   }
-  $role->save();
 }
 
 /**
@@ -651,7 +660,7 @@
   $request_time = \Drupal::time()->getRequestTime();
   foreach ($values as $field => $value) {
     // Set cookie for 365 days.
-    setrawcookie('Drupal.visitor.' . $field, rawurlencode($value), $request_time + 31536000, '/');
+    setrawcookie('Drupal.visitor.' . $field, rawurlencode($value), $request_time + 31536000, '/', '', TRUE, TRUE);
   }
 }
 
@@ -668,7 +677,7 @@
  */
 function user_cookie_delete($cookie_name): void {
   @trigger_error(__METHOD__ . '() is deprecated in drupal:11.4.0 and is removed from drupal:13.0.0. There is no replacement. See https://www.drupal.org/node/3581570', E_USER_DEPRECATED);
-  setrawcookie('Drupal.visitor.' . $cookie_name, '', \Drupal::time()->getRequestTime() - 3600, '/');
+  setrawcookie('Drupal.visitor.' . $cookie_name, '', \Drupal::time()->getRequestTime() - 3600, '/', '', TRUE, TRUE);
 }
 
 /**
