--- core/includes/install.core.inc	2026-05-15 11:55:29
+++ core/includes/install.core.inc	2026-05-15 12:27:58
@@ -390,7 +390,7 @@
   $install_state['database_verified'] = install_verify_database_settings($site_path);
   // A valid settings.php has database settings and a hash_salt value. Other
   // settings will be checked by \Drupal\system\Install\SystemRequirements.
-  $install_state['settings_verified'] = $install_state['config_verified'] && $install_state['database_verified'] && (bool) Settings::get('hash_salt', FALSE);
+  $install_state['settings_verified'] = $install_state['database_verified'] && (bool) Settings::get('hash_salt', FALSE);
 
   if ($install_state['settings_verified']) {
     try {
@@ -799,9 +799,9 @@
     'install_install_profile' => [],
     'install_import_translations' => [
       'display_name' => t('Set up translations'),
-      'display' => $needs_translations,
+      'display' => INSTALL_TASK_SKIP,
       'type' => 'batch',
-      'run' => $needs_translations ? INSTALL_TASK_RUN_IF_NOT_COMPLETED : INSTALL_TASK_SKIP,
+      'run' => INSTALL_TASK_SKIP,
     ],
     'install_configure_form' => [
       'display_name' => t('Configure site'),
@@ -856,7 +856,7 @@
   }
 
   // Now add any tasks defined by the installation profile.
-  if (!empty($install_state['parameters']['profile'])) {
+  if (isset($install_state['parameters']['profile'])) {
     // Load the profile install file, because it is not always loaded when
     // hook_install_tasks() is invoked (e.g. batch processing).
     $profile = $install_state['parameters']['profile'];
@@ -879,15 +879,15 @@
   $tasks += [
     'install_finish_translations' => [
       'display_name' => t('Finish translations'),
-      'display' => $needs_translations,
+      'display' => INSTALL_TASK_SKIP,
       'type' => 'batch',
-      'run' => $needs_translations ? INSTALL_TASK_RUN_IF_NOT_COMPLETED : INSTALL_TASK_SKIP,
+      'run' => INSTALL_TASK_SKIP,
     ],
     'install_finished' => [],
   ];
 
   // Allow the installation profile to modify the full list of tasks.
-  if (!empty($install_state['parameters']['profile'])) {
+  if (isset($install_state['parameters']['profile'])) {
     $profile = $install_state['parameters']['profile'];
     if ($install_state['profiles'][$profile]->load()) {
       $function = $install_state['parameters']['profile'] . '_install_tasks_alter';
