--- modules/captcha/captcha.module	2026-02-09 14:29:27
+++ modules/captcha/captcha1.module	2026-03-06 16:26:20
@@ -547,6 +547,7 @@
     $query = \Drupal::database()
       ->select('captcha_sessions', 'cs')
       ->fields('cs', ['solution'])
+      ->condition('status', CaptchaConstants::CAPTCHA_STATUS_UNSOLVED)
       ->condition('csid', $csid);
     // For non-"show always" persistence, prevent retrieval of solutions for
     // solved sessions to block replay attacks. Once a CAPTCHA is solved, its
