--- modules/account_field_split/account_field_split.install	2022-01-26 11:34:22.000000000 +0000
+++ modules/account_field_split/account_field_split.install	2022-01-26 11:34:50.000000000 +0000
@@ -9,7 +9,7 @@
  * Implements hook_install().
  */
 function account_field_split_install() {
-  module_set_weight('account_field_split', 101);
+  module_set_weight('account_field_split', 9);
   // Rebuild cache to ensure field are visible as splitted on account edit page.
   \Drupal::service('cache.discovery')->deleteAll();
   \Drupal::service('cache_tags.invalidator')->invalidateTags(['config:core.entity_form_display.user.user.default']);
@@ -30,5 +30,5 @@
  * Run the module alter after other modules.
  */
 function account_field_split_update_9001() {
-  module_set_weight('account_field_split', 101);
+  module_set_weight('account_field_split', 9);
 }
