--- core/modules/navigation/src/Plugin/Block/NavigationMenuBlock.php	2025-07-03 16:07:17
+++ core/modules/navigation/src/Plugin/Block/NavigationMenuBlock.php	2025-07-03 16:18:29
@@ -88,6 +88,7 @@
       ['callable' => 'menu.default_tree_manipulators:checkAccess'],
       ['callable' => NavigationMenuLinkTreeManipulators::class . ':addSecondLevelOverviewLinks'],
       ['callable' => 'menu.default_tree_manipulators:generateIndexAndSort'],
+      ['callable' => 'navigation.navigation_tree_manipulators:manipulate'],
     ];
     $tree = $this->menuTree->transform($tree, $manipulators);
     $build = $this->menuTree->build($tree);
--- core/modules/navigation/src/Hook/NavigationHooks.php	2025-07-03 16:19:53
+++ core/modules/navigation/src/Hook/NavigationHooks.php	2025-07-03 16:20:57
@@ -137,6 +137,8 @@
     $navigation_links = \Drupal::classResolver(NavigationContentLinks::class);
     assert($navigation_links instanceof NavigationContentLinks);
     $navigation_links->addMenuLinks($links);
+    //$navigation_links->removeAdminContentLink($links);
+    //$navigation_links->removeHelpLink($links);
   }
 
   /**
--- core/modules/navigation/navigation.services.yml	2025-07-03 16:23:37
+++ core/modules/navigation/navigation.services.yml	2025-07-03 16:35:53
@@ -48,3 +48,11 @@
 
   Drupal\navigation\Menu\NavigationMenuLinkTreeManipulators:
     autowire: true
+
+  navigation.navigation_tree_manipulators:
+    class: Drupal\navigation\NavigationMenuLinkTreeManipulators
+    arguments: ['@event_dispatcher']
+  Drupal\navigation\NavigationMenuLinkTreeManipulators: '@navigation.navigation_tree_manipulators'
+
+  navigation.event_subscriber:
+    class: Drupal\navigation\EventSubscriber\NavigationMenuLinkTreeSubscriber
